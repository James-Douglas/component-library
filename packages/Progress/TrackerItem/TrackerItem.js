import React from 'react';
import PropTypes from 'prop-types';

import ProgressIcon from '../ProgressIcon/ProgressIcon';

const TrackerItem = ({
  index, mobile, label, active, disabled,
}) => (
  <>
    <ProgressIcon index={index} active={active} disabled={disabled} mobile={mobile} />
    <div>{mobile ? label : ''}</div>
  </>
);

TrackerItem.propTypes = {
  index: PropTypes.number,
  mobile: PropTypes.bool,
  label: PropTypes.string,
  active: PropTypes.bool,
  disabled: PropTypes.bool,
};

TrackerItem.defaultProps = {
  index: 1,
  label: 'Label',
  mobile: false,
  active: false,
  disabled: false,
};

export default TrackerItem;
