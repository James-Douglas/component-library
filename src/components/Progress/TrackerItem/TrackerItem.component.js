import React from 'react';
import PropTypes from 'prop-types';

import ProgressIcon from '../ProgressIcon/ProgressIcon.component';
import styles from '../Tracker/styles';

const TrackerItem = ({
  index, mobile, label, active, disabled,
}) => (
  <>
    <style jsx>{styles}</style>
    <ProgressIcon index={index} active={active} disabled={disabled} mobile={mobile} />
    <div>{mobile ? label : ''}</div>
  </>
);

TrackerItem.propTypes = {
  index: PropTypes.number,
  mobile: PropTypes.bool,
  label: PropTypes.string,
  active: PropTypes.bool,
  disabled: PropTypes.bool,
};

TrackerItem.defaultProps = {
  index: 1,
  label: 'Label',
  mobile: false,
  active: false,
  disabled: false,
};

export default TrackerItem;
