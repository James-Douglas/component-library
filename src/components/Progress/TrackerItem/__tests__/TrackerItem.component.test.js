import React from 'react';
import { render } from '@testing-library/react';
import TrackerItem from '../TrackerItem.component';

describe('TrackerItem', () => {
  it('renders correctly', () => {
    const props = {
      index: 3,
      mobile: false,
      label: '66%',
      active: false,
      disabled: true,
    };
    const { container } = render(<TrackerItem index={props.index} mobile={props.mobile} label={props.label} active={props.active} disabled={props.disabled} />);
    expect(container.innerHTML).toMatchSnapshot();
  });
});
