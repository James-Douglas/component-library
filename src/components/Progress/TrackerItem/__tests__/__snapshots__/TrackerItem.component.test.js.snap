// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackerItem renders correctly 1`] = `"<style></style><span class=\\"progress-icon  disabled \\"><style></style>3</span><div></div>"`;
